### ISSUE TITLE
作者您好 感谢您提出的简单易懂的方法 但是我替换成我的数据集后程序loss说cnt=0 除数为0报错，所以我进行了改进，但是依然无法训练，请求帮助！

### ISSUE THREAD

**Issue by Astartboy** [MeloHo/superglue-with-pointnet#2]
The size of train data is 500
Loading processed data from data2/modelnet40_normal_resampled/airplane/train/modelnet_train_800pts_fps.dat...
The size of test data is 100
Loading processed data from data2/modelnet40_normal_resampled/airplane/test/modelnet_test_800pts_fps.dat...
Loaded checkpoint from epoch 4
epoch: 4, iteration: 0
epoch: 4, iteration: 1
epoch: 4, iteration: 2
epoch: 4, iteration: 3
epoch: 4, iteration: 4
epoch: 4, iteration: 5
epoch: 4, iteration: 6
epoch: 4, iteration: 7
epoch: 4, iteration: 8
epoch: 4, iteration: 9
epoch: 4, iteration: 10
epoch: 4, iteration: 10, train loss: nan
epoch: 4, iteration: 11
epoch: 4, iteration: 12

def loss_fn(T, gt_rotation, gt_translation, matches, matching_scores, point_set):
    """
    Compute Loss.
    
    Args:
        T (torch.Tensor): size = (B, 3, 4)
        gt_rotation (torch.Tensor): size = (B, 3, 3)
        gt_translation (torch.Tensor): size = (B, 3)
        matches (torch.Tensor): size = (B, n_points)
        matching_scores (torch.Tensor): size = (B, n_points)
        point_set (torch.Tensor): size = (B, 3, n_points)
        
    Return:
        Reprojection Error, cnt
    """
    
    gt_rotation = gt_rotation.float()
    gt_translation = gt_translation.float()
    
    B = T.shape[0]
    n_points = matches.shape[1]
    pred_rotation = T[:, :, :3]
    pred_translation = T[:, :, 3]
    
    # Compute re-projection loss
    loss_reproj = (torch.bmm(gt_rotation, point_set) + torch.unsqueeze(gt_translation, 2)) - (torch.bmm(pred_rotation, point_set) + torch.unsqueeze(pred_translation, 2))
    loss_reproj = torch.sum(torch.abs(loss_reproj)) / (B * n_points)
    
    # Compute inlier loss
    loss_inlier = 0
    cnt = 0
    
    for i in range(B):
        for j in range(n_points):
            if matches[i, j] == j:
                loss_inlier -= torch.log(matching_scores[i, j])
                cnt += 1
    if cnt == 0:
        return loss_reproj, 0, loss_reproj, 0
    else:
        return loss_reproj + 0.3 * loss_inlier / cnt, cnt / B, loss_reproj, loss_inlier / cnt
正常执行的话会报错 所以我加了这个if 但是还是训练不出来 loss一直是NAN

**Comment by Astartboy** [comment]
我想尝试对一个模型的特征提取效果 所以把数据集换成了600个airplane的第一个模型的随即变换，然后让这600个随机变换去训练 （也就是再变换后 分别为（600个第一个模型的变换）soucre和（变换的变换）target）这样子去理解的
